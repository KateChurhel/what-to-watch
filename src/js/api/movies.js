/* istanbul ignore file */
import { sendGetRequest } from './index';

export const getCategoryData = (category, data) => sendGetRequest(`/${category}/popular`, data);

export const getDetailsData = (category, id, data) => sendGetRequest(`/${category}/${id}`, data);

export const getSearchResult = (category, data) => sendGetRequest(`/search/${category}`, data);

export const getDiscoverResult = (data) => sendGetRequest('/discover/movie', data);

export const getGenres = () => sendGetRequest('/genre/movie/list');
