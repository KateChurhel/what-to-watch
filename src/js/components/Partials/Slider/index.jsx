// libraries
import React from 'react';
import PropTypes from 'prop-types';
import SwiperCore, { Navigation } from 'swiper';
import { Swiper, SwiperSlide } from 'swiper/react';
// styles
import 'swiper/swiper.scss';
import 'swiper/components/navigation/navigation.scss';
// helpers
import { formatCardData } from '../../../helpers/formatData';
// views
import SmallMovieCard from '../SmallMovieCard';

// install Swiper components
SwiperCore.use([Navigation]);

const Slider = ({ items = [], category }) => {
  const swiperSettings = {
    updateOnWindowResize: true,
    breakpoints: {
      1440: {
        slidesPerView: 5,
        slidesPerGroup: 5,
        spaceBetween: 60,
      },
      1024: {
        slidesPerView: 4,
        slidesPerGroup: 4,
        spaceBetween: 30,
      },
      480: {
        slidesPerView: 3,
        slidesPerGroup: 3,
        spaceBetween: 15,
      },
      0: {
        slidesPerView: 2,
        slidesPerGroup: 2,
        spaceBetween: 20,
      },
    },
    setWrapperSize: true,
    navigation: true,
  };

  return (
    <Swiper {...swiperSettings}>
      {items.map((item) => (
        <SwiperSlide key={`${category}-${item.id}`}>
          <SmallMovieCard {...formatCardData(item)} link={`/details/${category}/${item.id}`} />
        </SwiperSlide>
      ))}
    </Swiper>
  );
};

Slider.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
  })),
  category: PropTypes.string.isRequired,
};

export default Slider;
