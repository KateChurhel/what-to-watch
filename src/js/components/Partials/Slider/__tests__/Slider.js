import React from 'react';
import renderer from 'react-test-renderer';
import Slider from '../index';
import { MOCK_MOVIES } from '../../../../mocks/movies';

describe('Slider: ', () => {
  it('correctly renders', () => {
    const component = renderer.create(<Slider category="movie" items={MOCK_MOVIES.slice(0, 3)} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('correctly renders without data', () => {
    const component = renderer.create(<Slider category="movie" />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
