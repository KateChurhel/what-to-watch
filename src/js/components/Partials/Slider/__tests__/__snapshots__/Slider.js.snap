// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider:  correctly renders 1`] = `
<div
  className="swiper-container"
>
  <div
    className="swiper-button-prev"
  />
  <div
    className="swiper-button-next"
  />
  <div
    className="swiper-wrapper"
  >
    <div
      className="swiper-slide"
    >
      <a
        className="small-movie-card catalog__movies-card small-movie-card__link"
        href="/details/movie/313369"
      >
        <div
          className="small-movie-card__image "
        >
          <img
            alt="La La Land"
            height="450"
            loading="lazy"
            src="https://image.tmdb.org/t/p/w300/uDO8zWDhfWwoFdKS4fzkUJt0Rf0.jpg"
            width="300"
          />
        </div>
        <div
          className="small-movie-card__info"
        >
          <h3
            className="small-movie-card__title"
          >
            La La Land
          </h3>
        </div>
      </a>
    </div>
    <div
      className="swiper-slide"
    >
      <a
        className="small-movie-card catalog__movies-card small-movie-card__link"
        href="/details/movie/516486"
      >
        <div
          className="small-movie-card__image "
        >
          <img
            alt="Greyhound"
            height="450"
            loading="lazy"
            src="https://image.tmdb.org/t/p/w300/kjMbDciooTbJPofVXgAoFjfX8Of.jpg"
            width="300"
          />
        </div>
        <div
          className="small-movie-card__info"
        >
          <h3
            className="small-movie-card__title"
          >
            Greyhound
          </h3>
        </div>
      </a>
    </div>
    <div
      className="swiper-slide"
    >
      <a
        className="small-movie-card catalog__movies-card small-movie-card__link"
        href="/details/movie/385103"
      >
        <div
          className="small-movie-card__image "
        >
          <img
            alt="Scoob!"
            height="450"
            loading="lazy"
            src="https://image.tmdb.org/t/p/w300/jHo2M1OiH9Re33jYtUQdfzPeUkx.jpg"
            width="300"
          />
        </div>
        <div
          className="small-movie-card__info"
        >
          <h3
            className="small-movie-card__title"
          >
            Scoob!
          </h3>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`Slider:  correctly renders without data 1`] = `
<div
  className="swiper-container"
>
  <div
    className="swiper-button-prev"
  />
  <div
    className="swiper-button-next"
  />
  <div
    className="swiper-wrapper"
  />
</div>
`;
